exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -373,9 +376,9 @@ object SchemaUtils {
 
                     val incorrectCaseSensitiveName = existingCol.name.inProperCase() != col.nameUnquoted().inProperCase()
 
-                    val incorrectSizeOrScale = isIncorrectSizeOrScale(existingCol, columnType)
+                    val incorrectSizeOrScale = if (incorrectType) false else isIncorrectSizeOrScale(existingCol, columnType)
```

Comment:
The size and scale are only compared if the types are confirmed to be equivalent because the size and scale differences are taken into account when comparing the types. This might change later as we refine this feature.